polkadot/xcm/src/lib.rs

```diff
@@ -451,5 +671,56 @@ pub mod opaque {
 #[test]
 fn conversion_works() {
 	use latest::prelude::*;
-	let _: VersionedMultiAssets = (Here, 1u128).into();
+	let assets: Assets = (Here, 1u128).into();
+	let _: VersionedAssets = assets.into();
+}
+
+#[test]
+fn size_limits() {
+	extern crate std;
+
+	let mut test_failed = false;
+	macro_rules! check_sizes {
+        ($(($kind:ty, $expected:expr),)+) => {
+            $({
+                let s = core::mem::size_of::<$kind>();
+                // Since the types often affect the size of other types in which they're included
+                // it is more convenient to check multiple types at the same time and only fail
+                // the test at the end. For debugging it's also useful to print out all of the sizes,
+                // even if they're within the expected range.
+                if s > $expected {
+                    test_failed = true;
+                    std::eprintln!(
+                        "assertion failed: size of '{}' is {} (which is more than the expected {})",
+                        stringify!($kind),
+                        s,
+                        $expected
+                    );
+                } else {
+                    std::eprintln!(
```

Comment:
Not an error, so we should instead do this:

```suggestion
                    std::println!(
```